### PR TITLE
SSCS-11428 : Added hearing route to notifications

### PR DESCRIPTION
**Pull request by ss-evoco**
- Added new option to application.yaml to specify which hearing route should trigger a notification
- Updated all the tests to include hearing route
- Moved hearing booked and hearing reminder notifications to list assist oral route only

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by EliandroGV1 — APPROVED** [review]

**Review by StevensonSi — APPROVED** [review]

File: src/e2e/resources/jointParty/hearingReminderCallback.json [hunk 1]
```diff
@@ -237,6 +239,66 @@
...
+                  },
+                  "googleMapLink": "PUBLIC",
+                  "name": "PUBLIC"
+                }
+              }
+            }
+          }
+        ]
+      },
+      "hearings": {
```
Comment (DGordonV1) [inline src/e2e/resources/jointParty/hearingReminderCallback.json:272]: Flagging for future reference that this is duplicated.

File: src/main/java/uk/gov/hmcts/reform/sscs/config/NotificationConfig.java [hunk 2]
```diff
@@ -10,9 +10,7 @@
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.core.env.Environment;
 import org.springframework.stereotype.Component;
-import uk.gov.hmcts.reform.sscs.ccd.domain.Benefit;
-import uk.gov.hmcts.reform.sscs.ccd.domain.LanguagePreference;
-import uk.gov.hmcts.reform.sscs.ccd.domain.State;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
```
Comment (DGordonV1) [inline src/main/java/uk/gov/hmcts/reform/sscs/config/NotificationConfig.java:13]: I don't mind personally, but told we try not to use wildcards for imports.

**Review by DGordonV1 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/config/NotificationConfig.java (hunk 2 above)
Comment (ss-evoco) [reply src/main/java/uk/gov/hmcts/reform/sscs/config/NotificationConfig.java:13]: I saw that as well but im pretty sure I checked and with the changes ive added, I used all the classes within that package.

File: src/e2e/resources/jointParty/hearingReminderCallback.json (hunk 1 above)
Comment (ss-evoco) [reply src/e2e/resources/jointParty/hearingReminderCallback.json:272]: Thanks. I imagine this file will change drastically in the testing ticket work.

**Review by EliandroGV1 — APPROVED** [review]

**Review by StevensonSi — APPROVED** [review]

**Pull request merged**